### ISSUE TITLE
Expression key1 == key2 with different RSA public keys is evaluated True

### ISSUE THREAD

**Issue by jun66j5** [paramiko/paramiko#908]
There is 2 different RSA public keys from https://natmchugh.blogspot.jp/2015/09/md5-collisions-in-ssh-keys.html. Expression `key1 == key2` with the RSA public keys is evaluated `True`. It should be `False`.

```
>>> import paramiko
>>> import base64
>>> paramiko.__version__
'2.1.2'
>>> key1 = paramiko.RSAKey(data=base64.b64decode("""
... AAAAB3NzaC1yc2EAAAADAQABAAABAQD74ICRUoaAXkRwqGW51botLrFygSC1SN3C
... 7NR6cfFmpSdRuXZt6d20+GvAzOQKJpPTItK3YRjds1izapAqSM6o/BYWOikTKLHg
... zJ2s5Lwm6MZnPsnRJ9o9KbnXCytCrj3wFCcay0Re+TJrlPvXggFZlane2yMit2rW
... x/Fbly44wNzsJLIGTIhsz6UnfmR7Wi7GfiMZ2p48xn3rSHJ3lNPTOzWfg3PFgc2Y
... ftzUow1TQ5xaamyPRD/UOXBOj2pfCp75v3TWDwwq3qliBV6pxUhe/ou8ut1zaRET
... 5VTN4kFNziQZoxY4WVmCnCJxjBPt71hM9VzoujZggRNXct8uPdri
... """))
>>> key2 = paramiko.RSAKey(data=base64.b64decode("""
... AAAAB3NzaC1yc2EAAAADAQABAAABAQD74ICRUoaAXkRwqGW51botLrFygSC1SN3C
... 7NR6cfFmpSdRuXZt6d20+GvAzOQKJpPTItK3YRjds1izapCqSM6o/BYWOikTKLHg
... zJ2s5Lwm6MZnPsnRJ1o+KbnXCytCrj3wFCcaS0Re+TJrlPvXggFZlane2yMit2rW
... x/FbFy44wNzsJLIGTIhsz6UnfmR7Wi7GfiMZ2p68xX3rSHJ3lNPTOzWfg/PFgc2Y
... ftzUow1TQ5xaamyPRD/UOXBOj2pfCp75v3TWDwwq3qliBV6pxUhe/ou8ut1zaRET
... 5VTN4kFNziQZoxY4WVmCnCJxjBPt71hM9VzoujZggRNXct8uPdri
... """))
>>> key1.asbytes() == key2.asbytes()
False
>>> key1.get_fingerprint() == key2.get_fingerprint()
True
>>> key1 == key2
True
>>> hash(key1) == hash(key2)
True
```

I consider `PKey.__eq__` should compare `hash()` and `.asbytes()` between self and other.
```diff
diff --git a/paramiko/pkey.py b/paramiko/pkey.py
index c87daaed3..5866d18b3 100644
--- a/paramiko/pkey.py
+++ b/paramiko/pkey.py
@@ -101,7 +101,7 @@ class PKey(object):
         return cmp(self.asbytes(), other.asbytes())

     def __eq__(self, other):
-        return hash(self) == hash(other)
+        return hash(self) == hash(other) and self.asbytes() == other.asbytes()

     def get_name(self):
         """
```

**Comment by ploxiln** [comment]
note the function just above:

```python
    # noinspection PyUnresolvedReferences
    def __cmp__(self, other):
        """
        Compare this key to another.  Returns 0 if this key is equivalent to
        the given key, or non-0 if they are different.  Only the public parts
        of the key are compared, so a public key will compare equal to its
        corresponding private key.

        :param .Pkey other: key to compare to.
        """
        hs = hash(self)
        ho = hash(other)
        if hs != ho:
            return cmp(hs, ho)
        return cmp(self.asbytes(), other.asbytes())
```

`__cmp__()` is used to support for not-otherwise-implemented comparison special functions on python2 but is gone in python3 (as is `cmp()`)

**Comment by bitprophet** [comment]
Yea, feels like a missed spot in the Python 3 compat update. I feel like I have run into this before so one of us should look for prior art tickets.

**Comment by bitprophet** [comment]
This has been fixed as part of a related security patch - to be pushed and released soon. Credited in changelog - thx!

**Comment by andreif** [comment]
@bitprophet What version has introduced this regression? We have to stay on 2.7.2 so I am curious if that one was already affected? See https://github.com/pahaz/sshtunnel/issues/257

**Comment by andreif** [comment]
2.7.2 does have it https://github.com/paramiko/paramiko/blob/2.7.2/paramiko/pkey.py#L127-L140

**Comment by bskinn** [comment]
For future reference, this was fixed in [2.8.1](https://www.paramiko.org/changelog.html#2.8.1).
